src/components/date-picker-day/date-picker-day.scss

```diff
@@ -201,56 +204,52 @@
 :host([end-of-range][range-hover]) :not(.calcite--rtl) .day-wrapper,
 :host([start-of-range][range-hover]) .calcite--rtl .day-wrapper {
   background-image: linear-gradient(
-    to right,
+    to inline-end,
     var(--calcite-ui-foreground-current),
     var(--calcite-ui-foreground-current),
     var(--calcite-ui-foreground-2),
     var(--calcite-ui-foreground-2)
   );
   @apply rounded-none shadow-none;
+  &:hover {
+    background-image: linear-gradient(
+      to inline-end,
+      var(--calcite-ui-foreground-current),
+      var(--calcite-ui-foreground-current),
+      var(--calcite-ui-foreground-1),
+      var(--calcite-ui-foreground-1)
+    );
+    @apply rounded-none shadow-none;
+  }
 }
 
 :host([start-of-range][range-hover]) :not(.calcite--rtl) .day-wrapper,
 :host([end-of-range][range-hover]) .calcite--rtl .day-wrapper {
   background-image: linear-gradient(
-    to left,
+    to inline-start,
```

Comment:
`linear-gradient` doesn't support logical props (see https://github.com/Esri/calcite-components/pull/3636#issuecomment-989262308). This applies to all `linear-gradient`s below.